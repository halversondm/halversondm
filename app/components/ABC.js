/**
 * Created by Daniel on 6/26/2016.
 */
"use strict";

import React from "react";

const ABC = React.createClass({
  render() {
    return <div>
    </div>;
  }
});

export default ABC;
