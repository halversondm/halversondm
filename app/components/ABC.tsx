/**
 * Created by Daniel on 6/26/2016.
 */
import * as React from 'react'
import { Modal } from 'react-bootstrap'
import { CheckboxSeries } from './CheckboxSeries'
import { type ReactNode } from 'react'

export interface ABCState {
  antecedentOtherDisabled: boolean
  messages: string[]
  showModal: boolean
  user: UserState
}

export interface UserState {
  when: string
  antecedent: string
  antecedentOther: string
  location: string
  people: string[]
  peopleOther: string
  behavior: string[]
  behaviorOther: string
  duration: string
  intensity: string
  consequence: string[]
  consequenceOther: string
}

export class ABC extends React.Component<unknown, ABCState> {
  public state: ABCState
  private people: CheckboxSeries
  private behaviors: CheckboxSeries
  private consequences: CheckboxSeries

  private readonly antecedentValues: string[] = ['Given Direction/task, asked to do something',
    'Asked to wait', 'Difficulty with task/activity',
    'Preferred activity interrupted', 'Activity/Item denied ("told no")',
    'Loud, noisy environment', 'Given assistance / correction',
    'Transition between locations', 'Attention given to others',
    'Attention not given when wanted', 'Left alone (no indiv. attention)']

  private readonly locationValues: string[] = ['Home', 'School', 'Other']
  private readonly peopleValues: string[] = ['Mom', 'Dad', 'Sibling', 'Grandparents', 'Alone', 'Peers']
  private readonly behaviorValues: string[] = ['Refuse to follow directions', 'Makes verbal threats',
    'Grabbing/pulling', 'Crying/Whining', 'Screaming/Yelling', 'Scratching',
    'Biting', 'Spitting', 'Kicking', 'Flopping', 'Running Away',
    'Destroying property', 'Hitting Self', 'Hitting Others',
    'Verbal Refusal']

  private readonly durationValues: string[] = ['< 1 min', '1 - 5 min', '5 - 10 min', '10 - 30 min',
    '30 min - 1 hr', '1 - 2 hrs', '2 - 3 hrs', '3+ hrs']

  private readonly intensityValues: string[] = ['Low', 'Medium', 'High']
  private readonly consequenceValues: string[] = ['Verbal Redirection', 'Physical assist/prompt',
    'Ignored problem behavior', 'Kept on demand',
    'Verbal reprimand', 'Removed from activity',
    'Given a different activity/task', 'Lost Privilege', 'Sent to room',
    'Given a time out', 'Left alone']

  private readonly refHandlers = {
    peopleRef: (ref) => (this.people = ref),
    behaviorsRef: (ref) => (this.behaviors = ref),
    consequencesRef: (ref) => (this.consequences = ref)
  }

  constructor () {
    super({})
    this.state = this.initialState()
    this.close = this.close.bind(this)
    this.getTime = this.getTime.bind(this)
    this.antecedentRadios = this.antecedentRadios.bind(this)
    this.antecedentOtherText = this.antecedentOtherText.bind(this)
    this.locationRadios = this.locationRadios.bind(this)
    this.durationRadios = this.durationRadios.bind(this)
    this.intensityRadios = this.intensityRadios.bind(this)
    this.save = this.save.bind(this)
    this.postToServer = this.postToServer.bind(this)
    this.validSave = this.validSave.bind(this)
    this.reset = this.reset.bind(this)
  }

  initialState (): ABCState {
    return {
      antecedentOtherDisabled: true,
      messages: [],
      showModal: false,
      user: {
        when: '',
        antecedent: '',
        antecedentOther: '',
        location: '',
        people: [],
        peopleOther: '',
        behavior: [],
        behaviorOther: '',
        duration: '',
        intensity: '',
        consequence: [],
        consequenceOther: ''
      }
    }
  }

  close (): void {
    this.setState({ showModal: false })
  }

  getTime (): void {
    const user = this.state.user
    const time = new Date()
    user.when = time.toLocaleString()
    this.setState({ user })
  }

  antecedentRadios (event): void {
    const antecedent = event.target.value
    const user = this.state.user
    user.antecedent = antecedent
    let antecedentOtherDisabled = this.state.antecedentOtherDisabled
    if (antecedent === 'Other') {
      antecedentOtherDisabled = false
    } else {
      antecedentOtherDisabled = true
      user.antecedentOther = ''
    }
    this.setState({
      user,
      antecedentOtherDisabled
    })
  }

  antecedentOtherText (event): void {
    const antecedentOtherText = event.target.value
    const user = this.state.user
    user.antecedentOther = antecedentOtherText
    this.setState({ user })
  }

  locationRadios (event): void {
    const location = event.target.value
    const user = this.state.user
    user.location = location
    this.setState({ user })
  }

  durationRadios (event): void {
    const duration = event.target.value
    const user = this.state.user
    user.duration = duration
    this.setState({ user })
  }

  intensityRadios (event): void {
    const intensity = event.target.value
    const user = this.state.user
    user.intensity = intensity
    this.setState({ user })
  }

  save (): void {
    const user = this.state.user
    const peoplePull = this.people.pullCurrentState()
    user.people = peoplePull.selected
    user.peopleOther = peoplePull.otherLabelText
    const behaviorPull = this.behaviors.pullCurrentState()
    user.behavior = behaviorPull.selected
    user.behaviorOther = behaviorPull.otherLabelText
    const consequencePull = this.consequences.pullCurrentState()
    user.consequence = consequencePull.selected
    user.consequenceOther = consequencePull.otherLabelText
    this.setState({ user, showModal: true })
    if (this.validSave()) {
      this.postToServer()
    }
  }

  postToServer (): void {
    const data: string = JSON.stringify(this.state.user)
    console.log('ABC data sent ', data)
    const xhr: XMLHttpRequest = new XMLHttpRequest()
    xhr.open('POST', '/saveABC')
    xhr.setRequestHeader('Content-Type', 'application/json')
    xhr.onload = () => {
      const messages: string[] = []
      if (xhr.status >= 200 && xhr.status < 400) {
        console.log('succ ', xhr.responseText)
        messages.push(xhr.responseText)
      } else {
        console.log('unsucc ', xhr.responseText)
        messages.push(xhr.status + ' ' + xhr.statusText)
      }
      this.setState({ messages })
    }
    xhr.onerror = () => {
      console.log(xhr)
    }
    xhr.send(data)
  }

  validSave (): boolean {
    const messages: string[] = []
    if (this.state.user.people.length === 0) {
      messages.push('At least one Person is required to save.')
    } else if (this.state.user.people.indexOf('Other') === 0) {
      if (this.state.user.peopleOther === '') {
        messages.push(
          'For People - Other, the text description of Other must be entered.')
      }
    }
    if (this.state.user.behavior.length === 0) {
      messages.push('At least one Behavior is required to save.')
    } else if (this.state.user.behavior.indexOf('Other') === 0) {
      if (this.state.user.behaviorOther === '') {
        messages.push(
          'For Behavior - Other, the text description of Other must be entered.')
      }
    }
    if (this.state.user.consequence.length === 0) {
      messages.push('At least one Consequence is required to save.')
    } else if (this.state.user.consequence.indexOf('Other') === 0) {
      if (this.state.user.consequenceOther === '') {
        messages.push(
          'For Consequence - Other, the text description of Other must be entered.')
      }
    }
    if (this.state.user.antecedent.length === 0) {
      messages.push('An Antecedent is required to save.')
    } else if (this.state.user.antecedent === 'Other') {
      if (this.state.user.antecedentOther === '') {
        messages.push(
          'For Antecedent - Other, the text description of Other must be entered.')
      }
    }
    if (this.state.user.location.length === 0) {
      messages.push('A Location is required to save.')
    }
    if (this.state.user.duration.length === 0) {
      messages.push('A Duration is required to save.')
    }
    if (this.state.user.intensity.length === 0) {
      messages.push('An Intensity is required to save.')
    }
    if (this.state.user.when.length === 0) {
      messages.push('The date and time of the ABC is required to save.')
    }
    this.setState({ messages })
    return messages.length === 0
  }

  reset (): void {
    this.setState(this.initialState())
    this.people.reset()
    this.behaviors.reset()
    this.consequences.reset()
  }

  render (): ReactNode {
    return (
            <div>
                <h2 className="text-primary">ABC Checklist</h2>
                <div id="abcChecklist">
                    <div className="row">
                        <div className="col-md-1">
                            <h4>When?</h4>
                            <hr/>
                            <input className="btn btn-primary" onClick={this.getTime}
                                   type="button" value="Now" id="now"/>
                            <div id="dateTime">{this.state.user.when}</div>
                        </div>
                        <div className="col-md-4">
                            <h4>Antecedent</h4>
                            <hr/>
                            <b>What happened before the behavior?</b>
                            <br/>
                            {
                                this.antecedentValues.map((antecedent, i) => {
                                  return <div className="radio" key={i}>
                                        <label>
                                            <input type="radio" value={antecedent}
                                                   onClick={this.antecedentRadios}
                                                   checked={this.state.user.antecedent === antecedent}/>{antecedent}
                                        </label>
                                    </div>
                                })
                            }
                            <div className="radio">
                                <label>
                                    <input type="radio"
                                           checked={this.state.user.antecedent === 'Other'}
                                           value="Other"
                                           onClick={this.antecedentRadios}/> Other
                                </label>
                            </div>
                            <input type="text" className="form-control" id="antecedentOther"
                                   value={this.state.user.antecedentOther}
                                   placeholder="Enter Description"
                                   onChange={this.antecedentOtherText}
                                   disabled={this.state.antecedentOtherDisabled}/>
                            <br/> <b>Location</b>
                            <br/>
                            {
                                this.locationValues.map((location, i) => {
                                  return <div className="radio-inline"
                                                key={i}>
                                        <label>
                                            <input type="radio" onClick={this.locationRadios}
                                                   checked={this.state.user.location === location}
                                                   value={location}/>{location}
                                        </label>
                                    </div>
                                })
                            }
                            <br/> <b>People Present</b>
                            <br/>
                            <CheckboxSeries labels={this.peopleValues} selected={this.state.user.people}
                                            ref={this.refHandlers.peopleRef}
                                            otherLabelPlaceholder="Enter Another Person Present"
                                            otherLabelText={this.state.user.peopleOther}/>
                        </div>
                        <div className="col-md-3">
                            <h4>Behavior</h4>
                            <hr/>
                            <b>Select all that apply</b>
                            <CheckboxSeries labels={this.behaviorValues}
                                            ref={this.refHandlers.behaviorsRef}
                                            selected={this.state.user.behavior}
                                            otherLabelPlaceholder="Enter Description"
                                            otherLabelText={this.state.user.behaviorOther}/>
                            <br/> <b>Duration</b>
                            <br/>
                            {
                                this.durationValues.map((duration, i) => {
                                  return <div className="radio-inline" key={i}>
                                        <label>
                                            <input type="radio" onClick={this.durationRadios}
                                                   checked={this.state.user.duration === duration}
                                                   value={duration}/>{duration}
                                        </label>
                                    </div>
                                })
                            }
                            <br/> <b>Intensity</b>
                            <br/>
                            {
                                this.intensityValues.map((intensity, i) => {
                                  return <div className="radio-inline" key={i}>
                                        <label>
                                            <input type="radio" onClick={this.intensityRadios}
                                                   checked={this.state.user.intensity === intensity}
                                                   value={intensity}/>{intensity}
                                        </label>
                                    </div>
                                })
                            }
                        </div>
                        <div className="col-md-4">
                            <h4>Consequence</h4>
                            <hr/>
                            <b>What happened after?</b>
                            <br/>
                            <CheckboxSeries labels={this.consequenceValues}
                                            selected={this.state.user.consequence}
                                            ref={this.refHandlers.consequencesRef}
                                            otherLabelPlaceholder="Enter Description"
                                            otherLabelText={this.state.user.consequenceOther}/>
                        </div>
                    </div>
                    <br/>
                    <div className="form-inline">
                        <div className="form-group">
                            <input className="btn btn-success form-control" type="submit"
                                   onClick={this.save} value="Save" id="save"/>
                        </div>
                        <div className="form-group">
                            <input className="btn btn-danger form-control" type="reset"
                                   onClick={this.reset}
                                   value="Reset" id="reset"/>
                        </div>
                    </div>
                </div>
                <Modal show={this.state.showModal} onHide={this.close}>
                    <Modal.Header>
                        <p>ABC Save Results</p>
                    </Modal.Header>
                    <Modal.Body>
                        <ul>
                            {
                                this.state.messages.map((message, i) => {
                                  return <li key={i}>{message}</li>
                                })
                            }
                        </ul>
                    </Modal.Body>
                    <Modal.Footer>
                        <p style={{ right: 'auto' }}>Click anywhere to continue</p>
                    </Modal.Footer>
                </Modal>
                <div id="abcModal" className="modal">
                    <div className="modal-dialog" role="document">
                        <div className="modal-content">
                            <div className="modal-header">
                                <p>ABC Save Results</p>
                            </div>
                            <div className="modal-body">
                                <ul>
                                    {
                                        this.state.messages.map((message, i) => {
                                          return <li key={i}>{message}</li>
                                        })
                                    }
                                </ul>
                            </div>
                            <div className="modal-footer">
                                <p style={{ right: 'auto' }}>Click anywhere to continue</p>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
    )
  }
}
