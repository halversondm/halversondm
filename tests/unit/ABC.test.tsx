import React from "react";
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import ABC from "../../app/components/ABC";

describe("ABC App Testing", () => {
    it("Will have a snapshot that matches", () => {
        const component = render(<ABC/>);
        expect(component.container.toString()).toMatchSnapshot();
    });

    it("Will save a situation", async () => {
        const component = render(<ABC/>);
        await userEvent.click(document.getElementById("now"));
        await userEvent.click(screen.getByLabelText("Asked to wait"));
        await userEvent.click(screen.getByLabelText("Home"));
        await userEvent.click(screen.getByLabelText("Mom"));
        await userEvent.click(screen.getByLabelText("Refuse to follow directions"));
        await userEvent.click(screen.getByLabelText("Verbal Redirection"));
        await userEvent.click(screen.getByLabelText("5 - 10 min"));
        await userEvent.click(screen.getByLabelText("Medium"));
        await userEvent.click(document.getElementById("antecedentOtherRadio"));
        await userEvent.type(document.getElementById("antecedentOther"),"test antecedent other");
        await userEvent.click(document.getElementById("peopleOtherLabelCheckbox"));
        await userEvent.type(document.getElementById("peopleOtherLabelText"), "Dog");
        await userEvent.click(document.getElementById("behaviorOtherLabelCheckbox"));
        await userEvent.type(document.getElementById("behaviorOtherLabelText"), "test behavior other");
        await userEvent.click(document.getElementById("consequenceOtherLabelCheckbox"));
        await userEvent.type(document.getElementById("consequenceOtherLabelText"), "test consequence other");
        await userEvent.click(document.getElementById("save"));
    });

    it("Will save an empty situation", async () => {
        const component = render(<ABC/>);
        await userEvent.click(document.getElementById("save"));
    });
    it("Will save a partially empty situation", async () => {
        const component = render(<ABC/>);
        await userEvent.click(document.getElementById("antecedentOtherRadio"));
        await userEvent.click(document.getElementById("peopleOtherLabelCheckbox"));
        await userEvent.click(document.getElementById("behaviorOtherLabelCheckbox"));
        await userEvent.click(document.getElementById("consequenceOtherLabelCheckbox"));
        await userEvent.click(document.getElementById("save"));
    });
});
