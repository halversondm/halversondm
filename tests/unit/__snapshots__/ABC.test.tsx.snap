// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ABC App Testing > Will have a snapshot that matches 1`] = `"<div><div><h2 class="text-primary">ABC Checklist</h2><div id="abcChecklist"><div class="row"><div class="col-md-1"><h4>When?</h4><hr><input class="btn btn-primary" type="button" id="now" value="Now"><div id="dateTime"></div></div><div class="col-md-4"><h4>Antecedent</h4><hr><b>What happened before the behavior?</b><br><div class="radio"><label><input type="radio" value="Given Direction/task, asked to do something">Given Direction/task, asked to do something</label></div><div class="radio"><label><input type="radio" value="Asked to wait">Asked to wait</label></div><div class="radio"><label><input type="radio" value="Difficulty with task/activity">Difficulty with task/activity</label></div><div class="radio"><label><input type="radio" value="Preferred activity interrupted">Preferred activity interrupted</label></div><div class="radio"><label><input type="radio" value="Activity/Item denied (&quot;told no&quot;)">Activity/Item denied ("told no")</label></div><div class="radio"><label><input type="radio" value="Loud, noisy environment">Loud, noisy environment</label></div><div class="radio"><label><input type="radio" value="Given assistance / correction">Given assistance / correction</label></div><div class="radio"><label><input type="radio" value="Transition between locations">Transition between locations</label></div><div class="radio"><label><input type="radio" value="Attention given to others">Attention given to others</label></div><div class="radio"><label><input type="radio" value="Attention not given when wanted">Attention not given when wanted</label></div><div class="radio"><label><input type="radio" value="Left alone (no indiv. attention)">Left alone (no indiv. attention)</label></div><div class="radio"><label><input id="antecedentOtherRadio" type="radio" value="Other"> Other</label></div><input type="text" class="form-control" id="antecedentOther" placeholder="Enter Description" disabled="" value=""><br> <b>Location</b><br><div class="radio-inline"><label><input type="radio" value="Home">Home</label></div><div class="radio-inline"><label><input type="radio" value="School">School</label></div><div class="radio-inline"><label><input type="radio" value="Other">Other</label></div><br> <b>People Present</b><br><div class="checkbox"><label><input id="Mom" name="selectedLabel" type="checkbox" value="Mom"> Mom</label></div><div class="checkbox"><label><input id="Dad" name="selectedLabel" type="checkbox" value="Dad"> Dad</label></div><div class="checkbox"><label><input id="Sibling" name="selectedLabel" type="checkbox" value="Sibling"> Sibling</label></div><div class="checkbox"><label><input id="Grandparents" name="selectedLabel" type="checkbox" value="Grandparents"> Grandparents</label></div><div class="checkbox"><label><input id="Alone" name="selectedLabel" type="checkbox" value="Alone"> Alone</label></div><div class="checkbox"><label><input id="Peers" name="selectedLabel" type="checkbox" value="Peers"> Peers</label></div><div class="checkbox"><label><input id="peopleOtherLabelCheckbox" type="checkbox" name="peopleOtherLabelCheckbox" value="Other"> Other</label></div><input type="text" class="form-control" id="peopleOtherLabelText" placeholder="Enter Another Person Present" disabled="" value=""></div><div class="col-md-3"><h4>Behavior</h4><hr><b>Select all that apply</b><div class="checkbox"><label><input id="Refuse to follow directions" name="selectedLabel" type="checkbox" value="Refuse to follow directions"> Refuse to follow directions</label></div><div class="checkbox"><label><input id="Makes verbal threats" name="selectedLabel" type="checkbox" value="Makes verbal threats"> Makes verbal threats</label></div><div class="checkbox"><label><input id="Grabbing/pulling" name="selectedLabel" type="checkbox" value="Grabbing/pulling"> Grabbing/pulling</label></div><div class="checkbox"><label><input id="Crying/Whining" name="selectedLabel" type="checkbox" value="Crying/Whining"> Crying/Whining</label></div><div class="checkbox"><label><input id="Screaming/Yelling" name="selectedLabel" type="checkbox" value="Screaming/Yelling"> Screaming/Yelling</label></div><div class="checkbox"><label><input id="Scratching" name="selectedLabel" type="checkbox" value="Scratching"> Scratching</label></div><div class="checkbox"><label><input id="Biting" name="selectedLabel" type="checkbox" value="Biting"> Biting</label></div><div class="checkbox"><label><input id="Spitting" name="selectedLabel" type="checkbox" value="Spitting"> Spitting</label></div><div class="checkbox"><label><input id="Kicking" name="selectedLabel" type="checkbox" value="Kicking"> Kicking</label></div><div class="checkbox"><label><input id="Flopping" name="selectedLabel" type="checkbox" value="Flopping"> Flopping</label></div><div class="checkbox"><label><input id="Running Away" name="selectedLabel" type="checkbox" value="Running Away"> Running Away</label></div><div class="checkbox"><label><input id="Destroying property" name="selectedLabel" type="checkbox" value="Destroying property"> Destroying property</label></div><div class="checkbox"><label><input id="Hitting Self" name="selectedLabel" type="checkbox" value="Hitting Self"> Hitting Self</label></div><div class="checkbox"><label><input id="Hitting Others" name="selectedLabel" type="checkbox" value="Hitting Others"> Hitting Others</label></div><div class="checkbox"><label><input id="Verbal Refusal" name="selectedLabel" type="checkbox" value="Verbal Refusal"> Verbal Refusal</label></div><div class="checkbox"><label><input id="behaviorOtherLabelCheckbox" type="checkbox" name="behaviorOtherLabelCheckbox" value="Other"> Other</label></div><input type="text" class="form-control" id="behaviorOtherLabelText" placeholder="Enter Behavior Description" disabled="" value=""><br> <b>Duration</b><br><div class="radio-inline"><label><input type="radio" value="< 1 min">&lt; 1 min</label></div><div class="radio-inline"><label><input type="radio" value="1 - 5 min">1 - 5 min</label></div><div class="radio-inline"><label><input type="radio" value="5 - 10 min">5 - 10 min</label></div><div class="radio-inline"><label><input type="radio" value="10 - 30 min">10 - 30 min</label></div><div class="radio-inline"><label><input type="radio" value="30 min - 1 hr">30 min - 1 hr</label></div><div class="radio-inline"><label><input type="radio" value="1 - 2 hrs">1 - 2 hrs</label></div><div class="radio-inline"><label><input type="radio" value="2 - 3 hrs">2 - 3 hrs</label></div><div class="radio-inline"><label><input type="radio" value="3+ hrs">3+ hrs</label></div><br> <b>Intensity</b><br><div class="radio-inline"><label><input type="radio" value="Low">Low</label></div><div class="radio-inline"><label><input type="radio" value="Medium">Medium</label></div><div class="radio-inline"><label><input type="radio" value="High">High</label></div></div><div class="col-md-4"><h4>Consequence</h4><hr><b>What happened after?</b><br><div class="checkbox"><label><input id="Verbal Redirection" name="selectedLabel" type="checkbox" value="Verbal Redirection"> Verbal Redirection</label></div><div class="checkbox"><label><input id="Physical assist/prompt" name="selectedLabel" type="checkbox" value="Physical assist/prompt"> Physical assist/prompt</label></div><div class="checkbox"><label><input id="Ignored problem behavior" name="selectedLabel" type="checkbox" value="Ignored problem behavior"> Ignored problem behavior</label></div><div class="checkbox"><label><input id="Kept on demand" name="selectedLabel" type="checkbox" value="Kept on demand"> Kept on demand</label></div><div class="checkbox"><label><input id="Verbal reprimand" name="selectedLabel" type="checkbox" value="Verbal reprimand"> Verbal reprimand</label></div><div class="checkbox"><label><input id="Removed from activity" name="selectedLabel" type="checkbox" value="Removed from activity"> Removed from activity</label></div><div class="checkbox"><label><input id="Given a different activity/task" name="selectedLabel" type="checkbox" value="Given a different activity/task"> Given a different activity/task</label></div><div class="checkbox"><label><input id="Lost Privilege" name="selectedLabel" type="checkbox" value="Lost Privilege"> Lost Privilege</label></div><div class="checkbox"><label><input id="Sent to room" name="selectedLabel" type="checkbox" value="Sent to room"> Sent to room</label></div><div class="checkbox"><label><input id="Given a time out" name="selectedLabel" type="checkbox" value="Given a time out"> Given a time out</label></div><div class="checkbox"><label><input id="Left alone" name="selectedLabel" type="checkbox" value="Left alone"> Left alone</label></div><div class="checkbox"><label><input id="consequenceOtherLabelCheckbox" type="checkbox" name="consequenceOtherLabelCheckbox" value="Other"> Other</label></div><input type="text" class="form-control" id="consequenceOtherLabelText" placeholder="Enter Consequence Description" disabled="" value=""></div></div><br><div class="form-inline"><div class="form-group"><input class="btn btn-success form-control" type="submit" id="save" value="Save"></div><div class="form-group"><input class="btn btn-danger form-control" type="reset" id="reset" value="Reset"></div></div></div></div></div>"`;
